resources/assets/js/test/schemas/section-constraints.spec.json

```diff
@@ -0,0 +1,54 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"description": "Schema for documenting and validating section constraints",
+	"type": "object",
+	"properties": {
+		"allowedBlocks": {
+			"type": "array",
+			"items": {
+				"$ref": "#/definitions/blockName"
+			},
+			"minItems": 1
+		},
+		"defaultBlocks": {
+			"type": "array",
+			"items": {
+				"$ref": "#/definitions/blockName"
+			}
+		},
+		"allowedBlockGroups": {
+			"type": "array",
+			"items": {
+				"$ref": "#/definitions/blockGroup"
+			}
+		},
+		"min": {
+			"type": "number",
+			"minimum": 1,
+			"maximum": { "$data": "1/max" }
```

Comment:
$data references are supported by [AJV](https://github.com/epoberezkin/ajv#data-reference), which is probably the library we'd use to validate schemas.